### ISSUE TITLE
Tensor initialization with dtype and pytorch backend breaks

### ISSUE THREAD

**Issue by mganahl** [google/TensorNetwork#817]
`Tensor` initialization methods for pytorch currently break for this type of initialization:
```python 
t = tn.random_uniform((2,2), dtype=np.float64, backend='pytorch') 
```
(works for JAX and tensorflow)

**Comment by gaxler** [comment]
TensorFlow can accept Numpy types, but PyTorch can't. That's the reason this line fails for me. 

It works with dtype=torch.float64

**Comment by mganahl** [comment]
that's right, but it should be consistent

**Comment by Thenerdstation** [comment]
Eh, sounds more like a pytorch issue than an us issue. I assume you hit this from writing a test?

**Comment by mganahl** [comment]
that's right. I think it would be more convenient to use np-dtypes throughout for initialization, but I'm happy to leave it as is (though the error message that's thrown is quite cryptic as far as I remember, and should at least be fixed).

**Comment by gaxler** [comment]
How about initializing a numpy array and creating a tensor with torch.from_numpy?

**Comment by mganahl** [comment]
initialization of `Tensor` is currently done through backend functions (which I think is a good design). Those really should take dtypes matching the backend. We could add a backend function `convert_to_dtype` that takes a numpy dtype and converts it to the backend dtype; we then call it on the input dtype before passing it to the backend initialization function.

**Comment by alewis** [comment]
This is indeed the intended behaviour, like it or not (as it was for kron),
but I’m happy to change it. I don’t like silently converting dtypes if the
backend doesn’t though, and would suggest we use Martins solution of an
explicit conversion.

Adam

**Comment by alewis** [comment]
If you just want to write tests, the various tests in linalg/tests
demonstrate some workarounds.
